lib/config.atd

```diff
@@ -3,10 +3,14 @@ type prefix_rule <ocaml from="Rule"> = abstract
 type label_rule <ocaml from="Rule"> = abstract
 type project_owners_rule <ocaml from="Rule"> = abstract
 
+type pipeline = {
+  name: string;
+  ?failed_builds_channel: string nullable;
+}
+
 (* This type of rule is used for CI build notifications. *)
 type status_rules = {
-  ?allowed_pipelines : string list nullable; (* keep only status events with a title matching this list *)
-  ?failed_builds_channel: string nullable; (* channel to post failed builds notifications to *)
+  ?allowed_pipelines : pipeline list nullable; (* keep only status events with a title matching this list *)
```

Comment:
breaking change, will have to update the config in a bunch of different repos

Reply:
I have added this adapter to support the old format across repos with the old config format. This way we can gradually update the configs and then clean this up 

https://github.com/ahrefs/monorobot/pull/164/files#diff-25a0fd8d1dc5d6e0c1ac7fc3c8fcbba7d7a6a0a41a13f3d1d251173e514c52a4R71-R84

Reply:
please update the config documentation too